selfdrive/ui/sunnypilot/quietmode.py

```diff
@@ -0,0 +1,44 @@
+"""
+Copyright (c) 2021-, Haibin Wen, sunnypilot, and a number of other contributors.
+
+This file is part of sunnypilot and is licensed under the MIT License.
+See the LICENSE.md file in the root directory for more details.
+"""
+from cereal import car
+
+from openpilot.common.params import Params
+
+AudibleAlert = car.CarControl.HUDControl.AudibleAlert
+
+ALERTS_ALWAYS_PLAY = {
+  AudibleAlert.warningSoft,
+  AudibleAlert.warningImmediate,
+  AudibleAlert.promptDistracted,
+  AudibleAlert.promptRepeat,
+}
+
+class QuietDriveManager:
+  def __init__(self):
+    self.current_alert: int = car.CarControl.HUDControl.AudibleAlert.none
+    self.params = Params()
+    self.quiet_drive: bool = bool(self.params.get_bool("QuietDrive"))
+    self._frame = 0
+
+  def load_param(self) -> None:
+    self._frame += 1
+    if self._frame % 50 == 0:  # 2.5 seconds
+      self.quiet_drive = bool(self.params.get_bool("QuietDrive"))
+
+  def should_play_sound(self, current_alert: int) -> bool:
```

Comment:
Try to end the loop early by using the stock condition, so we are not coupling with our custom logic later in the loop, such as:

```py
if not self.enabled:
  # return with stock condition checks here
```